bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -978,6 +978,9 @@ private static boolean checkIsArrayType(BType sourceType, BArrayType targetType,
             if (sourceTupleType.getRestType() != null) {
                 tupleTypes.add(sourceTupleType.getRestType());
             }
+            if (tupleTypes.isEmpty()) {
```

Comment:
This will not work for something like

```ballerina
import ballerina/io;

public function main() {
    var x = [];

    any a = x;
    io:println(x is int[2]); // should be false
}
```

I think if `tupleTypes.isEmpty()` we should return true only if the target array state is `ArrayState.UNSEALED` or the target size is also 0.

Reply:
Fixed thanks.